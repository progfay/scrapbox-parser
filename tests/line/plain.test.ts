/* global describe it expect */

import '../jest-setup'

describe('plain', () => {
  it('Simple plain text', () => {
    expect('Plain text').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Plain text'
          }
        ]
      }
    ])
  })

  it('Blank line', () => {
    expect('').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: []
      }
    ])
  })

  it('Keep tail space', () => {
    expect('Tail space ->  ').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Tail space ->  '
          }
        ]
      }
    ])
  })
})
