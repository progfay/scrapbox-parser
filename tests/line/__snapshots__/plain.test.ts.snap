// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plain Blank line 1`] = `
[
  {
    "indent": 0,
    "nodes": [],
    "type": "line",
  },
]
`;

exports[`plain Keep tail space 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "Tail space ->  ",
        "text": "Tail space ->  ",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`plain Simple plain text 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "Plain text",
        "text": "Plain text",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;
